tools/org.eclipse.n4js.hlc/src/org/eclipse/jdt/internal/jarinjarloader/JIJConstants.java

```diff
@@ -1,9 +1,12 @@
 /*******************************************************************************
- * Copyright (c) 2007, 2009 IBM Corporation and others.
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
+ * Copyright (c) 2007, 2017 IBM Corporation and others.
```

Comment:
Do we really need to inline the code here?

Reply:
This is a big question and I have no idea. Ask @yoosiba :D It was done as part of making the headless compiler OSGI-capable. We should consider this question in a separate user story.

Reply:
@szarnekow do you mean whole `jdt/internal/jarinjarloader/*` ? Maybe not, but that would require experimentation. When creating this, it seemed like the easiest way to do it. Otherwise in the `pom.xml`, you need to fetch dependency which would contain _jarinjarloader_ and *while creating the hlc jar* make it use that code as entry point (https://github.com/eclipse/n4js/blob/master/tools/org.eclipse.n4js.hlc/pom.xml#L81-L109)

Also notice that the loader comes from `eclipse.jdt.ui` and I didn't want to include *UI* jar into *headless* compiler just to use 5 classes.

My long term plan was to just replace those copies with our own implementation... but priorities never aligned.
I recommend to do it now when switching. This code could be aligned more with the N4JS needs.

If you want to experiment with different approach, I have whole thing in this example project https://github.com/yoosiba/fhe I used it for creating this solution. It is a _bit_ smaller then whole N4JS, so iterations are faster.

Reply:
Hey @yoosiba, thank you for looking into this. The smaller project certainly helps to figure things out. @qtran-n4 could you please look into it in a subsequent PR?

Reply:
I have created a follow-up task: https://github.com/eclipse/n4js/issues/1300